skellytracker/trackers/openpose_tracker/openpose_recorder.py

```diff
@@ -0,0 +1,102 @@
+import json
+from typing import Union
+import numpy as np
+from pathlib import Path
+import re
+from tqdm import tqdm
+
+from skellytracker.trackers.base_tracker.base_recorder import BaseCumulativeRecorder
+
+
+class OpenPoseRecorder(BaseCumulativeRecorder):
+    def __init__(self, json_directory_path: Union[Path, str]):
+        super().__init__()
+        self.json_directory_path = Path(json_directory_path)
+
+    def extract_frame_index(self, filename: str) -> Union[int, None]:
+        """Extract the numeric part indicating the frame index from the filename."""
+        match = re.search(r"_(\d{12})_keypoints", filename)
+        return int(match.group(1)) if match else None
+
+    def parse_openpose_jsons(self, json_directory: Union[Path, str]) -> np.ndarray:
+        # Remove the iteration over subdirectories and focus on a single directory
+        json_directory = Path(json_directory)
+        files = list(Path(json_directory).glob("*.json"))
+        num_frames = len(files)
+        frame_indices = [self.extract_frame_index(f.name) for f in files]
+        frame_indices.sort()
```

Comment:
Currently `frame_indices` is a list of int and None, which is ambiguous to sort. Do we want any None values in `frame_indices` or do we want to do `[index for f in files if (index := self.extract_frame_index(f.name)) is not None]

Reply:
Great question honestly - this is probably one of the lines of code from the original openpose code, because I don't remember the motivation behind this 100%. 

Did you run it and see that it was a list of ints and Nones? I think my biggest question would honestly first be what's causing those None's, if you saw them

Reply:
The `extract_frame_index` method returns None is it doesn't find the frame index. So outside of "what's causing it to not find the index", we want to make sure the code path of "it doesn't find the index so it returns None" is covered. That could be by modifying the method to never return None (say by throwing an error if it doesn't find the frame index), or by making the list comprehension just skip over None values. It really depends on whether everything else depends on each file having a valid frame index is

Reply:
just fixed this